resources/adoptopenjdk_install.rb

```diff
@@ -1,22 +1,34 @@
 resource_name :adoptopenjdk_install
-include Java::Cookbook::Helpers
+include Java::Cookbook::AdoptOpenJdkHelpers
 default_action :install
 
-property :version, String, name_property: true
-property :variant, String, equal_to: %w(hotspot openj9 openj9-large-heap), default: 'openj9'
-property :java_home, String, default: lazy { "/usr/lib/jvm/java-#{version}-adoptopenjdk-#{variant}/#{sub_dir(url)}" }
-property :arch, default: lazy { node['kernel']['machine'] }
+property :version, String, name_property: true,
+  description: 'Java version to install'
+property :variant, String, equal_to: %w(hotspot openj9 openj9-large-heap), default: 'openj9',
+  description: 'Install flavour'
+
+property :url, String, default: lazy { default_adopt_openjdk_url(version) },
+  description: 'The URL to download from'
+property :checksum, String, regex: /^[0-9a-f]{32}$|^[a-zA-Z0-9]{40,64}$/, default: lazy { default_adopt_openjdk_checksum(version) }
+  description: 'The checksum for the downloaded file'
 
-property :url, String, default: lazy { default_adopt_openjdk_url(version)[variant] }
-property :checksum, String, regex: /^[0-9a-f]{32}$|^[a-zA-Z0-9]{40,64}$/, default: lazy { default_adopt_openjdk_checksum(version)[variant] }
-property :md5, String, regex: /^[0-9a-f]{32}$|^[a-zA-Z0-9]{40,64}$/
-property :java_home_mode, [Integer, String], default: '0755'
-property :bin_cmds, Array, default: lazy { default_adopt_openjdk_bin_cmds(version)[variant] }
+property :java_home, String, default: lazy { "/usr/lib/jvm/java-#{version}-adoptopenjdk-#{variant}/#{sub_dir(url)}" }
+  description: 'Set to override the java_home'
+property :java_home_mode, String, default: '0755',
+  description: 'The permission for the Java home directory'
```

Comment:
Wondering about property naming consistency, we have `java_home_mode` and then just `owner` and `group` below which are for java home. So either `java_home_owner` or change the other to `mode`

Reply:
home mode makes more sense here, you're right